src/main/java/games/strategy/engine/framework/ui/background/BackgroundTaskRunner.java

```diff
@@ -37,12 +37,13 @@ private BackgroundTaskRunner() {}
    */
   public static void runInBackground(
       final String message,
-      final Runnable backgroundAction)
+      final Runnable backgroundAction,
+      final GameRunner gameRunner)
```

Comment:
`BackgroundTaskRunner` is a good example of a type that probably shouldn't have a dependency on `GameRunner` (and I was the one who introduced it :smile:).  The dependency exists effectively to get a reference to the main frame window.  I think many of the types you had to change in this PR to accept a reference to a `GameRunner` instance are in this category.

A lot of this goes back a few months ago when we removed the `MainFrame` class (and thus the `MainFrame#getInstance()` method) and ended up adding several new UI-related methods to `GameRunner` so the main frame window reference could be encapsulated.  That wasn't necessarily a bad thing, and given what you're trying to achieve in this PR, having a static `MainFrame` instance would have to be addressed anyway.

However, I think this PR will require us to take a step back and re-think how a lot of this global state is managed.  One thing that's been bugging me is how `GameRunner` is some kind of _God Class_.  `GameRunner` is really an orchestration layer for the entire application that brings together many different types of services.  Those services should be extracted into interfaces, and instances of those interfaces passed to the downstream classes instead of `GameRunner` itself.

But let's start out with the low-hanging fruit like this class.  I'll submit a PR that refactors this class a bit so it doesn't depend on `GameRunner` (but its clients still will).  That may spark some further discussion on how to proceed.

Reply:
@ssoloff Assuming you're not going to build on top of this PR, should I just revert the last commit for now to avoid merge conflicts?
Otherwise if you want to build on top of this PR, go ahead and close it. I'd be happy to help you with it.

Reply:
Sorry, didn't see this comment before I submitted the PR.  Yeah, I didn't think the merge conflicts would be too big a deal (probably mostly changing `GameRunner` to `gameRunner`).

I'm thinking a lot of the changes to break dependencies should be done in `master` and then merged into this branch.  They are probably needed in general, and it will keep this PR from growing into a several-thousand line monstrosity. :smile: